PowerSimData2Switch.py

```diff
@@ -0,0 +1,432 @@
+                         dic_gen_age[grid.plant["type"].iloc[i]], 0, 0, 0, 
+                         dic_variable[grid.plant["type"].iloc[i]], 
+                         dic_baseload[grid.plant["type"].iloc[i]], 
+                         0, dic_fuel[grid.plant["type"].iloc[i]], 
+                         '.', '.', '.', '.', '.'])
+    
+        writer.writerow([''.join(["g", str(grid.plant.index[i])]), 
+                         grid.plant["type"].iloc[i], grid.plant["bus_id"].iloc[i], 0, 
+                         '.', 
+                         CalcHeatRate(grid.plant.index[i]), df_gen_vom.iloc[i], 
+                         dic_gen_age[grid.plant["type"].iloc[i]], 0, 0, 0, 
+                         dic_variable[grid.plant["type"].iloc[i]], 
+                         dic_baseload[grid.plant["type"].iloc[i]], 
+                         0, dic_fuel[grid.plant["type"].iloc[i]], 
+                         '.', '.', '.', '.', '.'])
+
+# write gen_build_predetermined.csv file.
+
+with open('gen_build_predetermined.csv', 'w', newline='') as gen_exist_file:
+    writer = csv.writer(gen_exist_file)
+    writer.writerow(["GENERATION_PROJECT", "build_year", "gen_predetermined_cap"])
+    
+    for i in range(len(grid.plant.index)):
+        writer.writerow([''.join(["g", str(grid.plant.index[i])]), 2019, 
+                         grid.plant['Pmax'].iloc[i]])
+
+# write gen_build_costs.csv file.
+
+with open('gen_build_costs.csv', 'w', newline='') as gen_buildcost_file:
+    writer = csv.writer(gen_buildcost_file)
+    writer.writerow(["GENERATION_PROJECT", "build_year", "gen_overnight_cost", "gen_fixed_om"])
+    
+    for i in range(len(grid.plant.index)):
+        writer.writerow([''.join(["g", str(grid.plant.index[i])]), 2019, 
+                         dic_geninvcost[grid.plant["type"].iloc[i]], 
+                         df_gen_fix_cost.iloc[i]])
+    
+    for i in range(inv_stage):
+        for j in range(len(grid.plant.index)):
+            writer.writerow([''.join(["g", str(grid.plant.index[j]), "i"]), inv_period[i], 
+                             dic_geninvcost[grid.plant["type"].iloc[j]], 
+                             df_gen_fix_cost.iloc[j]])
+        
+# write transmission_lines.csv file.
+# no circuit ID in PowerSimData.
+
+with open('transmission_lines.csv', 'w', newline='') as transmission_file:
+    writer = csv.writer(transmission_file)
+    writer.writerow(["TRANSMISSION_LINE", "trans_lz1", "trans_lz2", "trans_length_km", 
+                     "trans_efficiency", "existing_trans_cap"])
+
+    for i in range(len(grid.dcline.index)):
+        writer.writerow([''.join(["l", str(grid.dcline.index[i])]), 
+                         grid.dcline["from_bus_id"].iloc[i], grid.dcline["to_bus_id"].iloc[i], 
+                         GIS_distance2(grid.dcline["from_bus_id"].iloc[i], 
+                                      grid.dcline["to_bus_id"].iloc[i]), 0.99, 

```

Comment:
@YifanLi86 We are setting efficiencies of all dclines to be 0.99 here. Should we hard coded this for now or there is another mapping between voltages of dcline and efficiencies that could be added easily similarly as what we defined in `Branch_efficiency`?

Reply:
> @YifanLi86 We are setting efficiencies of all dclines to be 0.99 here. Should we hard coded this for now or there is another mapping between voltages of dcline and efficiencies that could be added easily similarly as what we defined in `Branch_efficiency`?

For MVP we can just do 0.99 for all dclines. For DC most of them the kV is pretty high, and DC cable loss is not as sensitive as AC cable towards kV level.